### ISSUE TITLE
Method cannot be called, for a protocol type

### ISSUE THREAD

**Issue by decorator-factory** [microsoft/pyright#7151]
Code ( [pyright playground link](https://pyright-play.net/?strict=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAqiApgGoCGIANFAArgxgDGYANgFDtOvkDOvUAGJgwACnphGLVgEoAXOyhKoAE2LAoAI0qjexVsBlQAtAD5MKGAuU2oAOgedufAQFk4wsNeVqN2kLr6hibmqFaKtkogxDAAriAoUAAsAEycBFAAvIQkFAEARAT5Mk48-FAAwmCW5KjEIN5KvlAA%2Bi1oMUgwxBBtgQY0ANbEcHKwJADaBAC6RmaEjbYOdk7VMLUo9VmVaxv1oiXswCLbLDV1IBOe00cidv4H7EA) ):
```py
from typing import TypeVar, Protocol

class Foo(Protocol):
    def bar(self) -> int:
        ...

T = TypeVar("T")

class Container:
    def __getitem__(self, key: type[T]) -> T:
        ...

container = Container()

foo = container[Foo]
foo.bar()
```
Unexpected error on the last line:
```
Method "bar" cannot be called because it is abstract and unimplemented
```

---

The actual code where I found the error is a bit more complex &mdash; it uses [aiohttp.web.AppKey](https://docs.aiohttp.org/en/stable/web_reference.html#aiohttp.web.AppKey) like so:
```
KEY_FOO = web.AppKey("foo", FooProtocol)
# KEY_FOO has type AppKey[FooProtocol]

# in create_app():
    app[KEY_FOO] = ConcreteFoo()

# in some HTTP handler:
    foo = app[KEY_FOO]
    foo.bar()
```

**Comment by erictraut** [comment]
Thanks for reporting the issue. This turns out to be the same issue as https://github.com/microsoft/pyright/issues/7105. It will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.349, which I just published.
